homeassistant/components/apsystems/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""The read-only binary sensors for APsystems local API integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from APsystemsEZ1 import ReturnAlarmInfo
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ApSystemsConfigEntry, ApSystemsData
+from .coordinator import ApSystemsDataCoordinator
+from .entity import ApSystemsEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class ApsystemsLocalApiBinarySensorDescription(BinarySensorEntityDescription):
+    """Describes Apsystens Inverter binary sensor entity."""
+
+    is_on: Callable[[ReturnAlarmInfo], bool | None]
+
+
+BINARY_SENSORS: tuple[ApsystemsLocalApiBinarySensorDescription, ...] = (
+    ApsystemsLocalApiBinarySensorDescription(
+        key="off_grid_status",
+        translation_key="off_grid_status",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        is_on=lambda c: bool(c.og),
+    ),
+    ApsystemsLocalApiBinarySensorDescription(
+        key="dc_1_short_circuit_error_status",
+        translation_key="dc_1_short_circuit_error_status",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        is_on=lambda c: bool(c.isce1),
+    ),
+    ApsystemsLocalApiBinarySensorDescription(
+        key="dc_2_short_circuit_error_status",
+        translation_key="dc_2_short_circuit_error_status",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        is_on=lambda c: bool(c.isce2),
+    ),
+    ApsystemsLocalApiBinarySensorDescription(
+        key="output_fault_status",
+        translation_key="output_fault_status",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        is_on=lambda c: bool(c.oe),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ApSystemsConfigEntry,
+    add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the binary sensor platform."""
+    config = config_entry.runtime_data
+
+    add_entities(
+        ApSystemsBinarySensorWithDescription(
+            data=config,
+            entity_description=desc,
+        )
+        for desc in BINARY_SENSORS
+    )
+
+
+class ApSystemsBinarySensorWithDescription(
+    CoordinatorEntity[ApSystemsDataCoordinator], ApSystemsEntity, BinarySensorEntity
+):
```

Comment:
You only have to inherit ApSystemsEntity

Reply:
If I do not inherit the `BinarySensorEntity` here, the tests fail and I get an unavailable state for all binary sensors.

Reply:
I *think* Joost referred to the `CoordinatorEntity` being unnecessary, because afaik ApSystemsEntity already inherits that

Reply:
No, `ApSystemsEntity` inherits only Entity from `homeassistant.helpers.entity`. If `CoordinatorEntity` is omitted, the binary sensors also cannot be set up successfully either.

Reply:
Okay, then my bad

Reply:
Ooh you're right

@mawoka-myblock A nice follow up would be to maybe create a second `CoordinatedApSystemsEntity` base entity so you can avoid the double inheritance